src/newgrf.cpp

```diff
@@ -2676,24 +2690,33 @@ static ChangeInfoResult TownHouseChangeInfo(uint hid, int numinfo, int prop, Byt
  * @param gvid ID of the global variable. This is basically only checked for zerones.
  * @param numinfo Number of subsequent IDs to change the property for.
  * @param buf The property value.
- * @param[in,out] translation_table Storage location for the translation table.
+ * @param gettable Function to get storage for the translation table.
  * @param name Name of the table for debug output.
  * @return ChangeInfoResult.
  */
 template <typename T>
-static ChangeInfoResult LoadTranslationTable(uint gvid, int numinfo, ByteReader &buf, std::vector<T> &translation_table, const char *name)
+static ChangeInfoResult LoadTranslationTable(uint gvid, int numinfo, ByteReader &buf, std::function<std::vector<T> &(GRFFile &)> gettable, std::string_view name)
 {
 	if (gvid != 0) {
 		GrfMsg(1, "LoadTranslationTable: {} translation table must start at zero", name);
 		return CIR_INVALID_ID;
 	}
 
+	std::vector<T> &translation_table = gettable(*_cur.grffile);
 	translation_table.clear();
 	translation_table.reserve(numinfo);
 	for (int i = 0; i < numinfo; i++) {
 		translation_table.push_back(T(BSWAP32(buf.ReadDWord())));
 	}
 
+	GRFFile *override = GetCurrentGRFOverride();
```

Comment:
`override` as a variable name is a bit ick (woe, context unaware syntax highlighting)